csharp/src/Drivers/BigQuery/BigQueryConnection.cs

```diff
@@ -764,22 +807,44 @@ private Field DescToField(BigQueryRow row)
         {
             Dictionary<string, string> metaData = new Dictionary<string, string>();
             metaData.Add("PRIMARY_KEY", "");
-            metaData.Add("ORDINAL_POSITION", row["ordinal_position"].ToString());
-            metaData.Add("DATA_TYPE", row["data_type"].ToString());
+            metaData.Add("ORDINAL_POSITION", GetValue(row["ordinal_position"]));
+            metaData.Add("DATA_TYPE", GetValue(row["data_type"]));
 
-            Field.Builder fieldBuilder = SchemaFieldGenerator(row["column_name"].ToString().ToLower(), row["data_type"].ToString());
+            Field.Builder fieldBuilder = SchemaFieldGenerator(GetValue(row["column_name"]).ToLower(), GetValue(row["data_type"]));
             fieldBuilder.Metadata(metaData);
 
-            if (!row["is_nullable"].ToString().Equals("YES", StringComparison.OrdinalIgnoreCase))
+            if (!GetValue(row["is_nullable"]).Equals("YES", StringComparison.OrdinalIgnoreCase))
             {
                 fieldBuilder.Nullable(false);
             }
 
-            fieldBuilder.Name(row["column_name"].ToString().ToLower());
+            fieldBuilder.Name(GetValue(row["column_name"]).ToLower());
 
             return fieldBuilder.Build();
         }
 
+        private string GetValue(object value)
+        {
+            switch (value)
+            {
+                case string sValue:
+                    if (string.IsNullOrEmpty(sValue))
+                        return string.Empty;
+                    return sValue;
+                default:
+                    if (value != null)
+                    {
+                        string? sValue = value.ToString();
+
+                        if (string.IsNullOrEmpty(sValue))
+                            return string.Empty;
```

Comment:
could be simplified to "return sValue ?? string.Empty;" though I think we probably shouldn't call it a success if ToString returns null.